### ISSUE TITLE
QoE Metrics Reporting - Support for RepSwitchList metric 

### ISSUE THREAD

**Issue by dsilhavy** [5G-MAG/rt-5gms-media-stream-handler#34]
# Feature description
Representation switch events should be captured by the `ExoPlayerListener` class and included in the QoE metrics report.

# Relevant specifications and corresponding clauses
* ISO/IEC 23009-1 5th edition - clause D.4.4
* 3GPP TS 26.501 Rel-17
* 3GPP TS 26.512 Rel-17
* 3GPP TS 26.247 Rel-17 clauses 10.6.2 and 10.6.3

# Example
````xml
<QoeMetric>
  <RepSwitchList>
    <RepSwitchEvent to="Rep2"/>
    <RepSwitchEvent to="Rep3"/> 
  </RepSwitchList>
</QoeMetric>
````

**Comment by rjb1000** [comment]
@dsilhavy: I would say that the `RepSwitchEvent` is pretty useless unless you also provide the optional `@t` timestamp attribute on that element. It would also be useful to provide the media presentation time at which the representation switch was made in the optional `@mt` attribute.

**Comment by dsilhavy** [comment]
@rjb1000 Yes that is how I implemented it, data looks like this:

````xml
<RepSwitchList>
    <RepSwitchEvent lto="-1" mt="268000" t="1687196338309" to="V300" />
    <RepSwitchEvent lto="-1" mt="268000" t="1687196338717" to="A48" />
</RepSwitchList>
````

The example above was taken from the 26.247 not sure why they omitted everything there expect `to`.

**Comment by rjb1000** [comment]
Very good.

My reading of ISO/IEC 23009-1 clause D.4.4 is that `@lto` is best omitted if there are no sub-representations.

**Comment by dsilhavy** [comment]
> Very good.
> 
> My reading of ISO/IEC 23009-1 clause D.4.4 is that `@lto` is best omitted if there are no sub-representations.

Thanks, agreed, unfortunately the XML serializer can not handle `null` values, so I had to set a default. My plan is to run over the final XML string at the end and then remove the optional/non-set attributes before dispatching the final XML to the Media Session Handler. As part of this step `lto` would be removed.

**Comment by rjb1000** [comment]
It sounds like your problems with XML output all stem from the framework you have chosen. Having to post-process the output to fix it up is a clear indication that something is badly wrong here.

- Are there better ways of using the framework that do not require post-processing?
- Is there a better framework available that you could switch to using?

**Comment by dsilhavy** [comment]
The [Jackson](https://github.com/FasterXML/jackson) library that is used now can handle `null` values and omits the attributes accordingly.

**Comment by dsilhavy** [comment]
Implemented in #67
